test/cases/parsing/issue-758/index.js

```diff
@@ -0,0 +1,50 @@
+it("should require existing module with supplied error callback", function(done) {
+	require.ensure(['./file'], function(){
+		var file = require('./file');
+		file.should.be.eql("file");
+		done();
+	}, function(error) {});
+});
+
+it("should call error callback on missing module", function(done) {
+	require.ensure(['./missingModule'], function(){
+		require('./missingModule');
+	}, function(error) {
+		error.should.be.instanceOf(Error);
+		error.message.should.be.eql('Cannot find module "./missingModule"');
+		done();
+	});
+});
+
+it("should call error callback on missing module in context", function(done) {
+	(function(module) {
+		require.ensure([], function(){
+			require('./' + module);
+		}, function(error) {
+			error.should.be.instanceOf(Error);
+			error.message.should.be.eql("Cannot find module './missingModule'.");
+			done();
+		});
+	})('missingModule');
+});
+
+it("should call error callback on exception thrown in loading module", function(done) {
+	require.ensure(['./throwing'], function(){
+		require('./throwing');
+	}, function(error) {
+		error.should.be.instanceOf(Error);
+		error.message.should.be.eql('message');
+		done();
+	});
+});
+
+it("should not call error callback on exception thrown in require callback", function(done) {
+	require.ensure(['./throwing'], function() {
+    require('./throwing');
+		throw new Error('message');
+	}, function(error) {
+		error.should.be.instanceOf(Error);
+		error.message.should.be.eql('message');
+		done();
+	});
+});
```

Comment:
I added a test case for this, but it's failing for 3 of the cases, for reasons I can't decipher.

Here's the test I added:

```
it("should call error callback when there is an error loading the chunk", function(done) {
	var temp = __webpack_require__.e;
	__webpack_require__.e = function() { return Promise.reject('fake chunk load error'); };
	require.ensure(['./file'], function(){
		var file = require('./file');
	}, function(error) {
		error.should.be.eql('fake chunk load error');
		done();
	});
	__webpack_require__.e = temp;
});
```

For some reason, this test passes for most of the times it is run, but fails 3 times for `TestCases hot parsing`, `TestCases hot-multi-step parsing`, and `TestCases all-combined parsing`. In those 3 instances, the callback is not getting executed, so the test times out. I'm a bit stumped as to why that might be.

@sokra any suggestions as to why the tests are failing for those 3 cases?